sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala

```diff
@@ -1040,6 +1040,14 @@ class ParquetSchemaSuite extends ParquetSchemaTest {
     }
   }
 
+  test("SPARK-40819 - ability to read parquet file with TIMESTAMP(NANOS, true)") {
+    val testDataPath = getClass.getResource("/test-data/timestamp-nanos.parquet")
+    val data = spark.read.parquet(testDataPath.toString).select("birthday")
+
+    assert(data.schema.fields.head.dataType == LongType)
+    assert(data.take(1).head.getAs[Long](0) == 1668537129000000000L)
```

Comment:
Shall we sort the read dataframe to be guarantee we compare the first row (Unless all rows have the same timestamp)?

Reply:
All rows have the same timestamp

Reply:
Can we have true nano second values to be sure they are not zeroed by the loader but loaded as in the file, e.g. `1665532812012345678L`.